autoendpoint/src/extractors/subscription.rs

```diff
@@ -308,7 +337,34 @@ fn validate_vapid_jwt(
                 return Err(VapidError::InvalidVapid(e.to_string()).into());
             }
             _ => {
-                metrics.clone().incr("notification.auth.bad_vapid.other");
+                // Attempt to match up the majority of ErrorKind variants.
+                // The third-party errors all defer to the source, so we can
+                // use that to differentiate for actual errors.
+                let mut tags = Tags::default();
+                let label = if e.source().is_none() {
+                    // These two have the most cardinality, so we need to handle
+                    // them separately.
+                    match e.clone().into_kind() {
+                        jsonwebtoken::errors::ErrorKind::InvalidRsaKey(_) => {
+                            "InvalidRsaKey".to_owned()
+                        }
+                        jsonwebtoken::errors::ErrorKind::MissingRequiredClaim(_) => {
+                            "MissingRequiredClaim".to_owned()
+                        }
+                        // NOTE: It's possible that any new error introduced could
+                        // be a source of cardinality. We should handle these as they
+                        // come up.
+                        _ => e.to_string(),
```

Comment:
There's still a few other variants that produce whitespace here such as `ErrorKind::Json`, can that and others still occur?